lib/node_modules/@stdlib/blas/ext/base/zfill/lib/ndarray.js

```diff
@@ -86,49 +84,50 @@ function zfill( N, alpha, x, stride, offset ) {
 	// Reinterpret the complex input array as a real-valued array:
 	view = reinterpret( x, 0 );
 
-	// Use loop unrolling if the stride is equal to `1`...
-	if ( stride === 1 ) {
+	// Adjust the stride and offset according to real-valued array:
+	ix = offsetX * 2;
+	strideX *= 2;
+
+	// Use loop unrolling if the stride is equal to `2`...
+	if ( strideX === 2 ) {
 		m = N % M;
 
 		// If we have a remainder, run a clean-up loop...
 		if ( m > 0 ) {
 			for ( i = 0; i < m; i++ ) {
-				j = ix * 2;
-				view[ j ] = re;
-				view[ j+1 ] = im;
-				ix += stride;
+				view[ ix ] = re;
+				view[ ix+1 ] = im;
+				ix += strideX;
 			}
 		}
 		if ( N < M ) {
 			return x;
 		}
 		for ( i = m; i < N; i += M ) {
-			j = ix * 2;
-			view[ j ] = re;
-			view[ j+1 ] = im;
-			view[ j+2 ] = re;
-			view[ j+3 ] = im;
-			view[ j+4 ] = re;
-			view[ j+5 ] = im;
-			view[ j+6 ] = re;
-			view[ j+7 ] = im;
-			view[ j+8 ] = re;
-			view[ j+9 ] = im;
-			view[ j+10 ] = re;
-			view[ j+11 ] = im;
-			view[ j+12 ] = re;
-			view[ j+13 ] = im;
-			view[ j+14 ] = re;
-			view[ j+15 ] = im;
-			ix += M;
+			view[ ix ] = re;
+			view[ ix+1 ] = im;
+			view[ ix+2 ] = re;
+			view[ ix+3 ] = im;
+			view[ ix+4 ] = re;
+			view[ ix+5 ] = im;
+			view[ ix+6 ] = re;
+			view[ ix+7 ] = im;
+			view[ ix+8 ] = re;
+			view[ ix+9 ] = im;
+			view[ ix+10 ] = re;
+			view[ ix+11 ] = im;
+			view[ ix+12 ] = re;
+			view[ ix+13 ] = im;
+			view[ ix+14 ] = re;
+			view[ ix+15 ] = im;
+			ix += M*2;
```

Comment:
@headlessNode Is there a reason why we shouldn't make `M = 16`? We could avoid the `*2` that way.

Reply:
I think we can go ahead and update here in order to shave off a FLOP.